pydantic/fields.py

```diff
@@ -124,16 +127,25 @@ def from_annotation(cls, annotation: type[Any] | _forward_ref.PydanticForwardRef
         >>>     foo: typing.Annotated[int, annotated_types.Gt(42)]
         >>>     bar: typing.Annotated[int, Field(gt=42)]
         """
+        final = False
+        if _typing_extra.is_finalvar(annotation):
+            final = True
+            if annotation is not typing_extensions.Final:
+                annotation = typing_extensions.get_args(annotation)[0]
+
         if _typing_extra.is_annotated(annotation):
             first_arg, *extra_args = typing_extensions.get_args(annotation)
+            if _typing_extra.is_finalvar(annotation):
```

Comment:
do we need this check too? looks like it duplicates the check above.

Reply:
I think this is necessary to handle both `Final[Annotated[int, Field(...)]]` and `Annotated[Final[int], Field(...)]`

Reply:
It looks like I have a typo and should be checking `if _typing_extra.is_finalvar(first_arg):` though

Reply:
<img width="437" alt="image" src="https://user-images.githubusercontent.com/35119617/230237825-47e08742-462b-4614-a0bb-cebe15918e52.png">
pycharm says we don't need to do `Annotated[Final[...` 👍

Reply:
this is interesting though:
<img width="432" alt="image" src="https://user-images.githubusercontent.com/35119617/230238062-e8c5eb07-d499-4eaa-bc05-6ae54dcb6ca4.png">
should we drop this functionality? haha. I guess no reason to given we've already ported it.

Reply:
agreed.

Reply:
let's keep it since we have it.